src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py

```diff
@@ -132,14 +134,14 @@ def send_uds_raw(
             else:
                 raise self.errors.ResponseNotReceivedError(msg_to_send)
 
-        resp_print = (
-            f"UDS response received {['0x{:02X}'.format(i) for i in resp]}"
-            if not isinstance(resp, bool)
-            else resp
-        )
-        log.internal_info(resp_print)
-        resp = UdsResponse(resp)
-        return resp
+        if log.isEnabledFor(logging.getLogger().level):
+            resp_print = (
+                f"UDS response received {['0x{:02X}'.format(i) for i in resp]}"
+                if not isinstance(resp, bool)
+                else resp
+            )
+            log.internal_info(resp_print)
+        return UdsResponse(resp)
```

Comment:
Wouldn't it make more sense to have a proper `__repr__` implemented in `UdsResponse`? 
You would then just do `log.internal_info(resp)`

Would be nice to have something like: 
```python
def __repr__(self):
    if self.is_negative:
        return f"NegativeUdsResponse({bytes(self.data).hex(sep=' ')}, nrc={self.nrc.name}})"
    return f"{self.__class__.__name__}({bytes(self.data).hex(sep=' ')})"

Reply:
Done